repository-type/repository-list-by-type/demo.js

```diff
@@ -0,0 +1,84 @@
+visualize({
+    auth: {
+        name: "joeuser",
+        password: "joeuser",
+        organization: "organization_1"
+    }
+}, function(v) {
+
+    var resourceTypes = ['folder',
+        'dataType',
+        'jdbcDataSource',
+        'awsDataSource',
+        'jndiJdbcDataSource',
+        'virtualDataSource',
+        'customDataSource',
+        'beanDataSource',
+        'xmlaConnection',
+        'listOfValues',
+        'file',
+        'reportOptions',
+        'dashboard',
+        'adhocDataView',
+        'query',
+        'olapUnit',
+        'reportUnit',
+        'domainTopic',
+        'semanticLayerDataSource',
+        'secureMondrianConnection',
+        'mondrianXmlaDefinition',
+        'mondrianConnection',
+        'inputControl'
+    ];
+
+    var sortBySelectElement = document.getElementById('sortBy');
+
+    var resourceTypeElement = document.getElementById('resourceType');
+
+    resourceTypes.forEach(function(i) {
+        let option = document.createElement('option');
+        option.text = i;
+        option.label = i;
+        resourceTypeElement.add(option)
+    });
```

Comment:
We have the `reportUnit` resource type option defined in the HTML file and the rest in the JavaScript file here. This is inconsistent and can be confusing. Please move all the resource type options to the HTML. In this case, the main part of Visualize.js API related to filtering by resource type will be more clear and easier to read and understand.